WordPress/src/main/java/org/wordpress/android/widgets/WPDialogSnackbar.java

```diff
@@ -0,0 +1,157 @@
+package org.wordpress.android.widgets;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.design.widget.Snackbar;
+import android.support.design.widget.Snackbar.SnackbarLayout;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.TextView;
+
+import org.wordpress.android.R;
+
+/**
+ * {@link Snackbar} with {@link android.app.Dialog}-like layout.  The layout views include title, message,
+ * positive button, and negative button.  Any empty or null view is hidden.  The only required view is message.
+ */
+public class WPDialogSnackbar {
+    private Snackbar mSnackbar;
+    private View mContentView;
+
+    private WPDialogSnackbar(@NonNull View view, @NonNull CharSequence text, int duration) {
+        switch (duration) {
```

Comment:
Can we use an enum for the duration? Since this is a private constructor it's not a big problem, but this forces the user of this class to know the implementation details as they'll be surprised if for example they pass in 2 seconds for the duration.

Reply:
> this forces the user of this class to know the implementation details as they'll be surprised if for example they pass in 2 seconds for the duration

Good point.  Rather than forcing the user to know the implementation details, we should just make it work like `Snackbar`.  That would mean replacing the entire `switch` statement with something like this.
```
mSnackbar = Snackbar.make(view, "", duration);
```
Do you think that's better than the `enum`?

Reply:
I checked the same thing first, but it looks like the set durations are `-2`, `-1` etc and because of that I didn't think it was better. Having said that, I think it's fine to just pass the duration and expect the caller to use the constants such as `Snackbar.LENGTH_INDEFINITE` since it's consistent with how Google does it and we are all used to it.

Reply:
Sounds good to me.  I updated the duration assignment in 18b0d67.